### PR TITLE
Linter update

### LINKED ISSUE #186: Change and Fix the Linter
**Description**
The actual linter [golint](https://github.com/golang/lint) of this repo has been deprecated see the reasons explain [here](https://github.com/golang/go/issues/38968).
I suggest that we find a new linter:
- [golangci-lint](https://golangci-lint.run/usage/quick-start)

And maybe a style checks like:
- [statis check](https://staticcheck.io/)
- [go vet](https://pkg.go.dev/cmd/vet)

I will close this issue in consequence:
- #185

In agreement with the Integration team, this will be done just after the release of Meilisearch v0.21.0

### PR DESCRIPTION
**Pull request by mmachatschek**
- Fixes #186
- Update meilisearch version to 0.21.1
- Update Go to supported versions

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ func TestClient_Version(t *testing.T) {
 			gotResp, err := tt.client.GetVersion()
 			require.NoError(t, err)
 			require.NotNil(t, gotResp, "Version() should not return nil value")
-			require.Equal(t, "0.21.0", gotResp.PkgVersion)
+			require.Equal(t, "0.21.1", gotResp.PkgVersion)
```
Comment (curquiza) [inline client_test.go:31]: Can you remove this test? We don't want the test to fail when a patch of MeiliSearch is released, like v0.21.1 🙂
@alallema do you agree with this?

Comment (mmachatschek) [reply client_test.go]: Done

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -41,16 +41,12 @@ Each PR should pass the tests and the linter to be accepted.
 docker pull getmeili/meilisearch:latest # Fetch the latest version of MeiliSearch image from Docker Hub
 docker run -p 7700:7700 getmeili/meilisearch:latest ./meilisearch --master-key=masterKey --no-analytics=true
 go test -v ./...
-# Install golint if needed (see comment below)
-go get -u golang.org/x/lint/golint
-# Use golint
-golint
+# Use golangci-lint
+docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:v1.42.0 golangci-lint run -v
```
Comment (alallema) [inline CONTRIBUTING.md:45]: Just a question why didn't use the `latest` instead of the `v1.42.0`?

**Review by alallema — APPROVED** [review]
Thanks, @mmachatschek for this PR!
LGTM

File: CONTRIBUTING.md (hunk 2 above)
Comment (mmachatschek) [reply CONTRIBUTING.md:45]: The CI step is also fixed to version 1.42.0. To mitigate differences in CI and local execution I fixed the version in the docker command to the same version

Comment (alallema) [reply CONTRIBUTING.md:45]: Perfect

**Comment by alallema** [comment]
bors merge

**Pull request merged**